### PR TITLE
Show empty deployments on overview if latest or in progress

### LINKED ISSUE #3363: Display empty deployment configs and latest deployments
The console UI should show at least the most recent deployment when scaled to 0 (so you know a deployment existed), and probably should show recent deployments (<N-1) for a brief period of time before hiding them.  There should probably be a way in the UI to see older deployments.  Eventually we'll want to have a scale control on 0 deployments.

Empty RCs not in deployment configs should always be shown.

### LINKED ISSUE #3474: Indicate a deployment is in progress
In the console, display empty rcs and also mark rcs as being deployed, failed, etc.

### LINKED ISSUE #5283: Cursor doesn't change to pointer when hovering over pod donut chart
~~This works fine in Firefox and Chrome.~~

This seems to happen on all platforms at narrower browser widths.

/cc @jwforres

### PR DESCRIPTION
**Pull request by spadgett**
Allow users to scale empty deployments back up from zero. Show when a deployment is in progress.

openshift_web_console

openshift_web_console

Fixes #3363
Fixes #3474
Fixes #5283

### CODE REVIEW

**Comment by spadgett** [comment]
@jwforres please review

File: assets/app/views/project.html [hunk 1]
```diff
@@ -195,8 +195,9 @@ <h2 ng-if="!displayRouteByService[serviceId]">
           </div>
 
           <!-- TODO implement filters for empty and present to return booleans for cases like this -->
-          <div ng-if="(podsByService[serviceId] | hashSize) == 0" class="osc-object components-panel deployment-block deployments none">
-            <span class="pficon pficon-info"></span> There are currently no pods for this service.
+          <div ng-if="(podsByService[serviceId] | hashSize) === 0 && (deploymentsByServiceByDeploymentConfig[serviceId] | hashSize) === 0"
```
Comment (jwforres) [inline assets/app/views/project.html]: Make sure this works when you have an RC that isn't a deployment and it is scaled to zero

Comment (spadgett) [inline assets/app/views/project.html]: What behavior do we want there?

Comment (spadgett) [inline assets/app/views/project.html]: Ah, let me fix. I misread the comment on RCs in the issue.

File: assets/app/scripts/directives/podStatusChart.js [hunk 2]
```diff
@@ -78,27 +85,42 @@ angular.module('openshiftConsole')
...
+            // Add a dummy group to draw an arc, which we style in CSS.
+            data.columns.push(["Empty", 1]);
+          } else {
+            // Unload the dummy group if present when there's real data.
+            data.unload = "Empty";
+          }
+
           if (!chart) {
-            config.data.columns = columns;
+            config.data.columns = data.columns;
```
Comment (jwforres) [inline assets/app/scripts/directives/podStatusChart.js]: should we also be setting config.data.unload here?

File: assets/app/scripts/directives/podStatusChart.js [hunk 3]
```diff
@@ -48,10 +48,11 @@ angular.module('openshiftConsole')
           type: "donut",
           bindto: '#' + $scope.chartId,
           donut: {
-            width: 10,
+            expand: false,
```
Comment (jwforres) [inline assets/app/scripts/directives/podStatusChart.js]: what behavior does this change?

File: assets/app/scripts/directives/podStatusChart.js (hunk 2 above)
Comment (spadgett) [inline assets/app/scripts/directives/podStatusChart.js]: No because it's a new chart, no data to unload.

Comment (jwforres) [inline assets/app/scripts/directives/podStatusChart.js]: Ah yep makes sense

File: assets/app/scripts/directives/podStatusChart.js (hunk 3 above)
Comment (spadgett) [inline assets/app/scripts/directives/podStatusChart.js]: It removes the effect that grows the arc on hover, which looked weird for the no pods case. We also don't handle clicks on the arcs anyway. The tooltip still displays.

**Comment by jwforres** [comment]
LGTM

**Comment by spadgett** [comment]
Empty RC:

openshift_web_console

Service with no deployments, replication controllers, or pods:

openshift_web_console

**Comment by spadgett** [comment]
> LGTM

Thanks. [merge]

**Comment by spadgett** [comment]
[test]

```
ok  	github.com/openshift/origin/pkg/generate/app	0.075s	coverage: 28.7% of statements
I1028 15:22:30.531703     676 newapp.go:605] Using "some/location.git" as the source for build
panic: test timed out after 1m0s
```

**Pull request merged**